test/inductor/test_compiled_optimizers.py

```diff
@@ -82,25 +86,36 @@ class KernelCounts(NamedTuple):
 }
 
 
-def build_compiled_opt_kwarg_db():
+def build_opt_kwarg_dbs():
     compiled_opt_db = []
+    disabled_opt_db = []
     for optim_info in optim_db:
-        if optim_info.optim_cls not in KERNEL_COUNTS:
-            continue
-
         for device in ["cpu", "cuda"]:
-            for optim_inputs in optim_info.optim_inputs_func(device):
-                for foreach in [True, False]:
-                    kwargs = dict(optim_inputs.kwargs)
-                    name = (
-                        f"test_{optim_info.optim_cls.__name__.lower()}"
-                        f"{'_foreach' if foreach else ''}"
+            for optim_inputs in _get_optim_inputs_including_global_cliquey_kwargs(
+                device, None, optim_info, skip=("fused", "differentiable")
+            ):
+                kwargs = dict(optim_inputs.kwargs)
+                # handle disabled optimizers
+                if optim_info.optim_cls not in KERNEL_COUNTS:
+                    kwargs["device"] = device
+                    disabled_opt_db.append(
+                        (
+                            optim_info.optim_cls,
+                            f"test_{optim_info.optim_cls.__name__.lower()}_disabled_falls_back_to_eager",
+                            kwargs,
+                        )
                     )
```

Comment:
Do we not do the same logic for disabled tests to save on CI cost/only run for one config?

Reply:
I don't think the CI cost is too high so FWIW I'd include them for coverage

Reply:
Yup, it doesn't really matter what the config is, the step is disabled so I don't think we need to test all configs.